RUFAS/output_manager.py

```diff
@@ -1624,28 +1624,59 @@ def _route_logs(self, log_pool: list[dict[str, str | dict[str, str]]]) -> None:
             A list of log, warning, and error dictionaries containing all the components needed
             to log the information to the appropriate pool.
         """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self.route_logs.__name__,
+        }
         for log in log_pool:
             if "error" in log:
                 if (
                     isinstance(log["error"], str)
                     and isinstance(log["message"], str)
                     and isinstance(log["info_map"], dict)
+                    and list(log.keys()) == ["error", "message", "info_map"]
                 ):
                     self.add_error(log["error"], log["message"], log["info_map"])
+                else:
+                    self.add_warning(
+                        "Wrong format for adding error.",
+                        f"Unable to add error with the format: {log}",
+                        info_map,
+                    )
             elif "log" in log:
                 if (
                     isinstance(log["log"], str)
                     and isinstance(log["message"], str)
                     and isinstance(log["info_map"], dict)
+                    and list(log.keys()) == ["log", "message", "info_map"]
                 ):
                     self.add_log(log["log"], log["message"], log["info_map"])
+                else:
+                    self.add_warning(
+                        "Wrong format for adding log.",
+                        f"Unable to add log with the format: {log}",
+                        info_map,
+                    )
             elif "warning" in log:
                 if (
                     isinstance(log["warning"], str)
                     and isinstance(log["message"], str)
                     and isinstance(log["info_map"], dict)
+                    and list(log.keys()) == ["warning", "message", "info_map"]
                 ):
                     self.add_warning(log["warning"], log["message"], log["info_map"])
+                else:
+                    self.add_warning(
+                        "Wrong format for adding warning.",
+                        f"Unable to add warning with the format: {log}",
+                        info_map,
+                    )
+            else:
+                self.add_warning(
+                    "Unsupported event key for output manager",
+                    f"Output manager can add logs, errors and warning. Given event contains the key {log.keys()}",
```

Comment:
```suggestion
                    f"Output manager can add logs, errors and warnings. Given event contains the key {log.keys()}",
```
Also curious why it's referred to here as an `event key` here? I don't believe we use that terminology anywhere in OM and it might be confusing to see this warning message.

I'm not sure how in depth you want to get with the warning message (like which of the keys of the log is incorrect) but I think it's good to provide the log.keys that was sent to this function and maybe have something saying the first key has to be `log`, `warning`, or `error`, the second key has to be `message`, and the third key has to be `info_map`).